### PR TITLE
[FabricClient] implement restart replica api

### PR DESCRIPTION
**Pull request by youyuanwu**
Implemented restart replica fabric client api for testing stateful service replica restart. This API will be used to test replica reopen workflow implementation.
Tested for the restarting of the example stateful service.

### CODE REVIEW

File: crates/libs/core/src/client/svc_mgmt_client.rs [hunk 1]
```diff
@@ -87,6 +108,20 @@ impl ServiceManagementClient {
         let res = ResolvedServicePartition::from_com(com);
         Ok(res)
     }
+
+    /// Simulates a service replica failure by restarting a persisted service replica,
+    /// closing the replica, and then reopening it. Use this to test your service for problems
+    /// along the replica reopen path. This helps simulate the report fault temporary path through client APIs.
+    /// This is only valid for replicas that belong to stateful persisted services.
+    pub async fn restart_replica(
```
Comment (rsajwani) [inline crates/libs/core/src/client/svc_mgmt_client.rs:116]: should we name it as restart_stateful_replica?

**Review by qbradley — APPROVED** [review]

File: crates/samples/echomain-stateful2/src/test.rs [hunk 2]
```diff
@@ -60,14 +60,23 @@ async fn test_partition_info() {
...
+    assert_ne!(stateful_replica.node_name, HSTRING::new());
+
+    let desc = RestartReplicaDescription {
+        node_name: stateful_replica.node_name.clone(),
+        partition_id: single.id,
+        replica_or_instance_id: stateful_replica.replica_id,
+    };
+    let mgmt = fc.get_service_manager();
+    mgmt.restart_replica(&desc, timeout).await.unwrap();
 }
```
Comment (rsajwani) [inline crates/samples/echomain-stateful2/src/test.rs:82]: should we have assert after restart replica

**Review by rsajwani — APPROVED** [review]

File: crates/libs/core/src/client/svc_mgmt_client.rs (hunk 1 above)
Comment (youyuanwu) [reply crates/libs/core/src/client/svc_mgmt_client.rs:116]: I prefer keep the same name as the COM api.

File: crates/samples/echomain-stateful2/src/test.rs (hunk 2 above)
Comment (youyuanwu) [reply crates/samples/echomain-stateful2/src/test.rs:82]: Once we have more apis ready I will add checks. The checks here needed is for query service health and status, which is not implemented yet.

**Pull request merged**